Disable access to old backups if storage is full UI changes

rustamshaikh opened the pull request:
UI changes to support the project **_Disable access to old backups when storage is full_** P2 project thread (peaFOp-yh-p2).

#### Proposed Changes

* Design P2: Disable access to old backups when storage is full – i2 Designs & redlines (p1HpG7-jKQ-p2)

#### Testing Instructions
- Test a site with normal usage <65%

<img width="803" alt="usage_normal" src="https://user-images.githubusercontent.com/13975226/213136408-44e6dbd7-f9a3-494b-8a37-3248ee43fcdf.png">

  - The help tooltip!
<img width="749" alt="usage_normal_with_tooltip" src="https://user-images.githubusercontent.com/13975226/213136439-456472f1-cc23-41f4-ba63-80d605cc8bb3.png">

- Test a site with moderate usage <80% and >=65%

<img width="750" alt="usage_warning" src="https://user-images.githubusercontent.com/13975226/213136964-2217df46-412e-406e-80b7-f2f19548a0c7.png">

- Test a site with critical usage <100% and >=80%

<img width="764" alt="usage_critical" src="https://user-images.githubusercontent.com/13975226/213137161-65439bb9-cf33-4aa9-972c-380b78816385.png">

- Test a site with full usage (Note: The below screenshots are of Calypso Blue!)
   1) When the site uses all the storage within the initial 7 days
   
   <img width="771" alt="usage_full_within_6_days" src="https://user-images.githubusercontent.com/13975226/213137688-b5f7f39d-eb73-4288-a8e7-4f44fa079fee.png">

   2)  When the site uses all the storage within its plan retention days and we discarded some backups up to the maximum extent (currently up to the last 7 days)
   <img width="746" alt="usage_full_7_days_blue" src="https://user-images.githubusercontent.com/13975226/213138157-070d5b04-0fff-4b49-b314-a9a53c3399cc.png">
   
- Test a site that uses all of the storage and then we discarded some backups but not up to the maximum extent 

<img width="762" alt="usage_discarded" src="https://user-images.githubusercontent.com/13975226/213138685-e7384d2e-916d-466f-86d2-7a3471f5ff47.png">


Unit tests: 
```
$ yarn test-client client/components/backup-storage-space/test
 PASS  client/components/backup-storage-space/test/hooks.js
 PASS  client/components/backup-storage-space/test/storage-usage-levels.js

Test Suites: 2 passed, 2 total
Tests:       31 passed, 31 total
Snapshots:   0 total
Time:        1.143 s
Ran all test suites matching /client\/components\/backup-storage-space\/test/i.
```


#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Asana Task: [Update banner "Storage usage" with the new info: num of backups and new texts](1203465157135577-as-1203558009266227)

rustamshaikh committed:
Initial commit

- Updated storage usage messages
- Updated conditions to show them
- Added informations like days of backups saved, minimum days of backups allowed.
- Updated backup storage component's styles
- Added selectors to get allowed current and minimum days of backups, backups saved.

rustamshaikh committed:
Updated test cases for getUsageLevel

Added/updated test cases to check for correct usage level returned when the new fields `daysOfBackupsAllowed`, `minDaysOfBackupsAllowed` and `planRetentionDays` are passed.

rustamshaikh committed:
Refactored code for better maintainability and as per guidelines

- Separated storage help tooltip into its own file.
- Refactored CSS as per the changes and guide
- Moved dynamic backup saved text to hooks file to keep parent clean and simple.

rustamshaikh committed:
Added changes to show icon/text with red when storage is full

rustamshaikh committed:
Some UI and data fixes

- Fixes progress-bar color when some backups are discarded message is displayed
- Fixed tooltip icon position when displayed on Calypso Blue UI
- Fixed storage full message to show correct no. of backups saved (instead of showing min. days of backups allowed)

rustamshaikh committed:
Added more tests!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7753916

Thank you @rustamshaikh for including a screenshot in the description! This is really helpful for our translators.

Initsogar commented on client/components/backup-storage-space/storage-usage-levels.ts:
```diff
@@ -20,14 +21,30 @@ const THRESHOLD_VALUES = Object.keys( THRESHOLDS )
 
+	if ( !! minDaysOfBackupsAllowed && !! daysOfBackupsAllowed && !! planRetentionDays ) {
+		// if current allowed days of backups is equal to the minimum, return storage full.
+		if ( minDaysOfBackupsAllowed === daysOfBackupsAllowed ) {
```
Should we validate the site used storage on this condition too?

Initsogar reviewed (approved):
Looks good to me 👍🏽
Just left a comment, maybe an improvement that can be addressed in a new PR.

rustamshaikh committed:
Added addtional conditions while checking for storage full

- Updated the condition to rely on minDaysOfBackupsAllowed >= 			minDaysOfBackupsAllowed >= daysOfBackupsSaved and usedStorage >= available
- Updated unit tests

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7753916).

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7753916).

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7753916).
